include/roboteam_ai/analysis/play-utilities/Play.h

```diff
@@ -15,27 +15,34 @@ namespace rtt::ai::analysis {
  *
  * TODO: Should this object have a behaviourtree associated to it? I think yes
  */
-class Play {
-   public:
-    Play();
-
-    Play(std::string name, std::vector<std::function<bool(world::World*, world::Field*)>> invariants);
-    void setInvariants(const std::vector<std::function<bool(world::World*, world::Field*)>>& invariants);
-    const std::vector<std::function<bool(world::World*, world::Field*)>>& getInvariants() const;
-
-    /**
-     *
-     * @return true if all the invariants of this strategy are true
-     */
-    bool isValidPlay(rtt::ai::world::World* world, rtt::ai::world::Field* field);
-
-    std::string getName();
-
-   protected:
-    std::vector<std::function<bool(world::World*, world::Field*)>> invariants;
-    std::shared_ptr<bt::BehaviorTree> tree;
-    std::string name;
-};
+    class Play {
+    public:
+        Play() = default;
+
+        /**
+         * @return true if all the invariants of this strategy are true
+         */
+        bool isValidPlay(rtt::ai::world::World *world, rtt::ai::world::Field *field);
+        // TODO: Move this to the derived class
+        /**
+         * Returns a score based on how fitting this play is given a world and field state (currently hardcoded, should be moved to derived classes)
+         * @param world the current world
+         * @param field the current field
+         * @return a score between 0 and 10, 10 being the best
```

Comment:
I would suggest not using a maximum in the future, this makes it easier to extend existing code rather than having to rescale a lot of current scores every time you add a new play.

Reply:
While I agree with you that this way is not optimal, I am not sure how else we can do this. I think until we get a good idea on how to fix it this is okay to keep though